src/main/java/reactor/netty/http/server/HttpServer.java

```diff
@@ -238,6 +240,42 @@ public final HttpServer forwarded(boolean forwardedEnabled) {
 		}
 	}
 
+	/**
+	 * Specifies whether support for the {@code "HAProxy proxy protocol"}
+	 * for deriving information about the address of the remote peer is enabled.
+	 *
+	 * @param proxyProtocolEnabled if true support for the {@code "HAProxy proxy protocol"}
+	 *                         for deriving information about the address of the remote peer is enabled,
+	 *                         otherwise disabled.
+	 * @return a new {@link HttpServer}
+	 */
+	public final HttpServer proxyProtocol(boolean proxyProtocolEnabled) {
+		if (proxyProtocolEnabled) {
+			try {
+				Class.forName("io.netty.handler.codec.haproxy.HAProxyMessageDecoder");
+			}
+			catch (ClassNotFoundException e) {
+				throw new MissingDependencyException(
```

Comment:
Can we just use UnsupportedOperationException to avoid a new public type for now ?